kong/plugins/ip-restriction/handler.lua

```diff
@@ -29,6 +31,28 @@ do
 end
 
 
+local is_http_subsystem = ngx.config.subsystem == "http"
+
+
+local do_exit
+if is_http_subsystem then
+  do_exit = function(status, message)
+    return kong.response.error(status, message)
+  end
+
+else
+  do_exit = function(status, message)
+    local tcpsock, err = ngx_req.socket(true)
+    if err then
+      error(err)
+    end
+
+    tcpsock:send(message)
```

Comment:
We use that for debugging, But I can see issues with protocols like ssh.
Getting rid of the message would also break all of the tests we added.
@hbagdi how would you like us to proceed?

Reply:
@chronolaw @hanshuebner Opinions?
Why not only close the socket and do nothing more?
(please hold until we arrive at a solution)

Reply:
@hbagdi I agree that sending a message through the TCP channel is unhygienic.  Furthermore, one could argue that from a security perspective, it may be undesirable to reveal the fact that there is a black- or whitelist that causes the closure of the connection.  From a usability perspective, however, I feel that sending a message before closing the connection is very helpful in debugging.  Making the message sending configurable would be an option that'd also help with testability.  Another option will be to change the tests so that instead of looking for the in-band rejection message, they will look at the log file.

Personally, I'll be fine keeping things how they are.

Reply:
I vote for keeping the message, but am happy to do whatever.  @hbagdi who is going to make the final call?  Please let me know how to proceed.  Would like like to wrap this up, we've been working on this for quite a while and are currently having to use a custom build for now.  Thanks.

Reply:
Mixing up strings on an arbitrary TCP connection doesn't seem the right thing to do unless someone can prove enough value for doing the wrong thing.

@hanshuebner Could you please check if Kong does something similar on a TCP connection either in the core or plugin? I don't recall anything like that.
If this is a net new thing, let's remove the message and merge this. If a message is needed, we will do an opt-in flag on the plugin configuration for it.

Reply:
I could not find any evidence that we do such in-band error signaling on TCP sockets at this point.

@scrudge Can you remove the message and update your tests so that they look at specific log entries instead?  This is done in several other tests as well, here's an example: https://github.com/Kong/kong/blob/96eb92f2e301b4b249eae710d51715931a683656/spec/02-integration/05-proxy/06-ssl_spec.lua#L270

Reply:
I refactored to use kong respnse error (which does not return the message on TCP connections.  Also added a log message at level warn when a connection is denied.  Updated the test to use the log message to validate the deny.

Reply:
@hanshuebner @hbagdi 
Do you mind reviewing my changes?
Thanks!